sdk/client/client.go

```diff
@@ -196,13 +198,7 @@ func (r *Response) Unmarshal(model interface{}) error {
 
 		// Reassign the response body as downstream code may expect it
 		r.Body = io.NopCloser(bytes.NewBuffer(respBody))
-	}
-
-	if strings.Contains(contentType, "application/octet-stream") {
-		if _, ok := model.([]byte); !ok {
-			return fmt.Errorf("internal-error: `model` must be []byte but got %+v", model)
-		}
-
+	case isContentType("application/octet-stream") || strings.HasPrefix(contentType, "text/"):
```

Comment:
We parse this from the Swagger into a `RawFile` object: https://github.com/hashicorp/pandora/blob/a912fd797a6f61a032649cb4239cf716c442e4a6/data/Pandora.Definitions.ResourceManager/Automation/v2022_08_08/DscConfiguration/Operation-GetContent.cs#L16-L27

Which [when output in the Go SDK](https://github.com/hashicorp/pandora/blob/a912fd797a6f61a032649cb4239cf716c442e4a6/tools/sdk/resourcemanager/models.go#L83-L84) is output as `[]byte` - so I think we can simplify this to:

```go
if strings.Contains(contentType, "application/octet-stream") || strings.Contains(contentType, "text/powershell") {
```

and then re-use the `application/octet-stream` logic? Should we need to parse additional types in the future we can handle that, but we _shouldn't_ be doing so since the Importer will normalize this for us - so we can handle just the simple case for now

---

We'll also need to add similar support to the Marshal function, since the `ReplaceContent` operation also sends a `RawFile`: https://github.com/hashicorp/pandora/issues/2185

Reply:
thanks! I have jsut upadted this PR. as i found the generated response Model field is a `*[]byte` 
https://github.com/hashicorp/go-azure-sdk/blob/bd4de95c1ffb70323a9dec602b723a3bf0dc908f/resource-manager/automation/2022-08-08/dscconfiguration/method_getcontent_autorest.go#L15-L18

and it pass a `&result.Model` (is a `**[]byte` now) to Marshal. so i handled both `*[]byte` and `**[]byte` in this PR.
https://github.com/hashicorp/go-azure-sdk/blob/bd4de95c1ffb70323a9dec602b723a3bf0dc908f/resource-manager/automation/2022-08-08/dscconfiguration/method_getcontent_autorest.go#L64